thread_pthread_mn.c

```diff
@@ -589,20 +676,43 @@ timer_thread_register_waiting(rb_thread_t *th, int fd, enum thread_sched_waiting
         }
         else {
             VM_ASSERT(fd >= 0);
+#if HAVE_SYS_EVENT_H
+            EV_SET(&ke[num_events], fd, EVFILT_WRITE, EV_ADD, 0, 0, (void *)th);
+            num_events++;
+#else
             epoll_events |= EPOLLOUT;
+#endif
         }
     }
 
     rb_native_mutex_lock(&timer_th.waiting_lock);
     {
+#if HAVE_SYS_EVENT_H
+        if (num_events > 0) {
+            if (kevent(timer_th.event_fd, ke, num_events, NULL, 0, NULL) == -1) {
```

Comment:
I'm not familiar with `kevent` system call but is there any failure we can ingore?

Reply:
In other words `kevent` should be success any time? (otherwize is it a BUG?)

Reply:
```ruby
io1 = open("/tmp/x", 'w')
io2 = IO.for_fd(io1.fileno)
p io1.close #=> nil
p io2.close #=> Bad file descriptor (Errno::EBADF)
```

I think `EBADF` is possible.